Update instructions to build SourceKit-LSP on Windows (#1742)
swiftlang/sourcekit-lsp: CONTRIBUTING.md

### Windows

> To build SourceKit-LSP on Windows, swift-syntax needs to be built as dynamic libraries so it does not exceed the maximum symbol limit in a single binary. Additionally, the equivalent search paths to the linux build need to be passed. Run the following in Command Prompt.

Reviewer: "swift-syntax needs to be built as dynamic libraries" reads a little weird to me, how about:

```suggestion
To build SourceKit-LSP on Windows, the swift-syntax libraries need to be built as dynamic libraries so we do not exceed the maximum symbol limit in a single binary. Additionally, the equivalent search paths to the linux build need to be passed. Run the following in Command Prompt.
```
